### DISCUSSION TITLE
Contributions à un projet Open source par des étudiants de l'université Paris-Saclay

### DISCUSSION THREAD

**Discussion by rami-pro** [MTES-MCT/dialog#1132] [Ideas]
Bonjour,

Nous sommes un groupe de trois étudiants en Master MIAGE à l'Université de Paris-Saclay, et dans le cadre de notre formation, nous sommes encouragés à contribuer à des projets open source. Après avoir consulté le vôtre, nous avons constaté qu'il répond à nos critères : une bonne qualité de code, un projet actif, et, en plus, un projet français.

Nous serions ravis de pouvoir apporter notre contribution en matière de développement, tests, CI/CD, amélioration de la documentation pour rendre le projet plus accessible, ainsi qu'en participant à des sessions de pair programming avec l'un de vos ingénieurs. Si vous avez des issues ouvertes, nous serions également heureux que vous puissiez nous en attribuer afin de nous permettre de démarrer rapidement.

**Comment by MathieuFV** [comment]
Bonjour ! Merci de vous être intéressés à notre projet ! On a bien vu votre message et on est en train de regarder comment on pourrait travailler ensemble :)

Si vous voulez qu'on échange par mail ou qu'on se fasse une visio, envoyez nous vos coordonnées à dialog@beta.gouv.fr

**Comment by rami-pro** [comment]
Bonjour @MathieuFV,

Je vous ai recontacté par mail à l'adresse indiquée, mais il est possible que mon message ait atterri dans les spams ou que vous ne l'ayez pas vu. Je tenais donc à réitérer l'enthousiasme de notre équipe à contribuer à votre projet. Nous serions ravis d’organiser une visio ou de poursuivre nos échanges à l'adresse suivante : pro.rami.gb@gmail.com.

**Comment by florimondmanca** [comment]
Bonjour @rami-pro 

Si je peux me permettre, je pense que la meilleure façon de procéder serait peut-être de parcourir vous-mêmes la liste d'issues, et voir si certains vous semblent à votre portée ?

Dans un projet open source, cela fonctionne généralement comme cela, de façon très aysnchrone.

La petite difficulté étant néanmoins que jusqu'ici nous rédigions les tickets pour l'équipe et pas forcément à destination de contributeurs extérieur, il peut donc manquer du contexte.

Si un ticket vous semble abordable mais que vous avez besoin d'un peu plus de contexte, n'hésitez pas à :

* Parcourir le dossier de documentation interne https://github.com/MTES-MCT/dialog/tree/main/docs
* Poser des questions complémentaires sur le ticket

---

Par ailleurs, si vous avez des langages de prédilection ou si vous utilisez des frameworks, n'hésitez pas à vous rapprocher de ces gros projets qui proposent souvent de nombreuses issues de type "easy picker" mais aussi plus avancées. Par exemple si vous êtes dans l'écosystème Python, des projets comme [django](https://github.com/django/django), [spaCy](https://github.com/explosion/spaCy), etc.

Les projets type "bibliothèque" ou "framework" sont de très bons candidats pour contribuer à l'open source, alors que ça me paraît un peu moins aisé pour les projets "produit final" comme DiaLog qui nécessitent un important contexte métier.
